static/js/components/CopyButton.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { FC, useState } from "react";
+import { Tooltip, Button } from "@canonical/react-components";
+import { Position } from "@canonical/react-components/dist/components/Tooltip/Tooltip";
+
+interface Props {
+  text: string;
+  className: string;
+  tooltipPosition?: Position;
+}
+
+const CopyButton: FC<Props> = ({
+  text,
+  className,
+  tooltipPosition = "left",
+}) => {
+  const [copied, setCopied] = useState(false);
+
+  const handleCopy = async () => {
+    try {
+      await navigator.clipboard.writeText(text);
+      setCopied(true);
+
+      setTimeout(() => {
+        setCopied(false);
```

Comment:
This is a bit odd. It changes the tooltip message and the icon back, but why not keep it as it is without the timeout / revert function?

Reply:
It's taken from anbox dashboard with no modifications. Since it was in main I assumed it was reviewed and good already?

Reply:
I might have missed it there.

Reply:
I think it's good as it is now. The user would get confused with no revert as they might want to copy it again and would not know that the button is back to the "can-copy" state. Also, it is a common way to handle this kind of functionality, I've seen it everywhere done like this.

Reply:
ok